mobile/library/cc/engine_builder.cc

```diff
@@ -255,37 +250,55 @@ EngineBuilder& EngineBuilder::enableDnsCache(bool dns_cache_on) {
 
 EngineBuilder& EngineBuilder::addStringAccessor(std::string name,
                                                 StringAccessorSharedPtr accessor) {
-  bootstrapIncompatible();
   string_accessors_[std::move(name)] = std::move(accessor);
   return *this;
 }
 
 EngineBuilder& EngineBuilder::addNativeFilter(std::string name, std::string typed_config) {
-  bootstrapIncompatible();
   native_filter_chain_.emplace_back(std::move(name), std::move(typed_config));
   return *this;
 }
 
 EngineBuilder& EngineBuilder::addPlatformFilter(std::string name) {
-  bootstrapIncompatible();
-  platform_filters_.push_back(std::move(name));
-  return *this;
-}
-
-EngineBuilder& EngineBuilder::setUseBootstrap() {
-  ENVOY_BUG(!config_bootstrap_incompatible_, "Config not compatible with bootstrap builder");
-  use_bootstrap_ = true;
+  addNativeFilter(
+      "envoy.filters.http.platform_bridge",
+      absl::StrCat(
+          "{'@type': "
+          "type.googleapis.com/envoymobile.extensions.filters.http.platform_bridge.PlatformBridge, "
+          "platform_filter_name: ",
+          name, "}"));
   return *this;
 }
 
 std::string EngineBuilder::generateConfigStr() const {
+  if (!config_override_for_tests_.empty()) {
+    return config_override_for_tests_;
+  }
+
+  std::string preresolve_hostnames = "[";
+  bool first = true;
+  for (auto& hostname : dns_preresolve_hostnames_) {
+    absl::StrAppend(&preresolve_hostnames, (first ? "" : ","), "{address: ", hostname,
+                    ", port_value: 443}");
+    first = false;
+  }
+  absl::StrAppend(&preresolve_hostnames, "]");
+
+  std::string virtual_clusters = "[";
+  first = true;
+  for (auto& cluster : virtual_clusters_) {
+    absl::StrAppend(&virtual_clusters, (first ? "" : ","), cluster);
+    first = false;
+  }
```

Comment:
Just throwing this out there as an idea, avoids the hard-to-read ternary expression:

```suggestion
  std::string comma = "";
  for (auto& hostname : dns_preresolve_hostnames_) {
    absl::StrAppend(&preresolve_hostnames, comma, "{address: ", hostname,
                    ", port_value: 443}");
    comma = ",";
  }
  absl::StrAppend(&preresolve_hostnames, "]");

  std::string virtual_clusters = "[";
  comma = "";
  for (auto& cluster : virtual_clusters_) {
    absl::StrAppend(&virtual_clusters, comma, cluster);
    comma = ",";
  }
```